apps/src/aiTutor/views/AITutorSuggestedPrompts.tsx

```diff
@@ -24,38 +26,86 @@ const AITutorSuggestedPrompts: React.FunctionComponent = () => {
 
   const level = useAppSelector(state => state.aiTutor.level);
 
-  const sources = useAppSelector(state => state.javalabEditor.sources);
+  // For PythonLab
+  const pythonLabSource = useAppSelector(
+    state => state.lab2Project?.projectSources?.source
+  );
+
+  const hasPythonLabError = useAppSelector(state => state.lab2System.hasError);
+  const hasRunOrTestedPythonLabCode = useAppSelector(
+    state => state.lab2System.hasRun || state.lab2System.hasValidated
+  );
+  const isPythonLabRunning = useAppSelector(
+    state => state.lab2System.isRunning
+  );
+  const isPythonLabValidating = useAppSelector(
+    state => state.lab2System.isValidating
+  );
+  const {hasConditions, satisfied} = useAppSelector(
+    state => state.lab.validationState
+  );
+  const pythonLabValidationPassed = !hasConditions || satisfied;
+
+  // For JavaLab
+  const javaLabSources = useAppSelector(state => state.javalabEditor.sources);
   const fileMetadata = useAppSelector(
     state => state.javalabEditor.fileMetadata
   );
   const activeTabKey = useAppSelector(
     state => state.javalabEditor.activeTabKey
   );
-  const studentCode = sources[fileMetadata[activeTabKey]].text;
 
-  const hasCompilationError = useAppSelector(
+  const hasJavaLabCompilationError = useAppSelector(
     state => state.javalabEditor.hasCompilationError
   );
-  const hasRunOrTestedCode = useAppSelector(
+  const hasRunOrTestedJavaLabCode = useAppSelector(
     state => state.javalab.hasRunOrTestedCode
   );
-  const isRunning = useAppSelector(state => state.javalab.isRunning);
-  const validationPassed = useAppSelector(
+  const isJavaLabRunning = useAppSelector(state => state.javalab.isRunning);
+  const javaLabValidationPassed = useAppSelector(
     state => state.javalab.validationPassed
   );
 
-  const dispatch = useAppDispatch();
+  function getOptionsByLabType(labType: string) {
+    if (labType === 'Pythonlab') {
+      const studentCode =
+        typeof pythonLabSource !== 'string' && pythonLabSource
+          ? getActiveFileForSource(pythonLabSource)?.contents || ''
+          : '';
+      // Only show a suggested prompt if we aren't currently running or validating code,
+      // code has been run or validated, and we aren't waiting for a chat response.
+      const showOption =
+        !isPythonLabRunning &&
+        !isPythonLabValidating &&
+        hasRunOrTestedPythonLabCode &&
+        !isWaitingForChatResponse;
+      const showGenericErrorOption = showOption && hasPythonLabError;
+      const showValidationOption = showOption && !pythonLabValidationPassed;
```

Comment:
Sorry for the late comment! If a user runs the code, but didn't yet try validating the code, 'Why are my tests failing?' is displayed. Maybe we could wait to show this suggested prompt after they attempt to validate code?